File: 2026/bit_manipulation/construct_min_bitwise_array_i_3314.py
class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        result = []
        for num in nums:
            found = False
            for x in range(num):
                if (x | (x + 1)) == num:
                    result.append(x)
                    found = True
                    break
            if not found:
                result.append(-1)
        return result



class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        result = []
        for num in nums:
            if num == 2:
                result.append(-1)
                continue
            found = False
            for j in range(1, 32):
                if (num & (1 << j)) > 0:
                    continue
                result.append(num ^ (1 << (j - 1)))
                found = True
                break
            if not found:
                result.append(-1)
        return result
        

File: 2026/bit_manipulation/construct_min_bitwise_array_i_3314.cpp
//T.C : O(n * m), m = maximum number in nums
//S.C : O(1)
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
        vector<int> result;
        for (int& num : nums) {
            bool found = false;
            for (int x = 0; x < num; x++) {
                if ((x | (x + 1)) == num) {
                    result.push_back(x);
                    found = true;
                    break;
                }
            }
            if (!found) {
                result.push_back(-1);
            }
        }
        return result;
    }
};

//T.C : O(n)
//S.C : O(1)
class Solution {
public:
    vector<int> minBitwiseArray(vector<int>& nums) {
        vector<int> result;


        for(int &num : nums) {
            if(num == 2) {
                result.push_back(-1);
                continue;
            }

            bool found = false;
            for(int j = 1; j < 32; j++) {
                if((num & (1 << j)) > 0) { //set bit
                    continue;
                }

                //we found an unset bit at jth position
                result.push_back((num ^ (1 << (j-1)))); //set the first set bit to 0
                found = true;
                break;
            }

            if(!found)
                result.push_back(-1);
        }

        return result;
    }
};
